src/query/v2/plan/vertex_count_cache.hpp

```diff
@@ -54,31 +55,14 @@ class VertexCountCache {
     return 1;
   }
 
-  bool LabelIndexExists(storage::v3::LabelId label) { return request_router_->IsPrimaryLabel(label); }
+  bool LabelIndexExists(storage::v3::LabelId label) { return PrimaryLabelExists(label); }
 
-  bool LabelPropertyIndexExists(storage::v3::LabelId /*label*/, storage::v3::PropertyId /*property*/) { return false; }
-
-  std::vector<std::pair<query::v2::Expression *, query::v2::plan::FilterInfo>> ExtractPrimaryKey(
-      storage::v3::LabelId label, std::vector<query::v2::plan::FilterInfo> property_filters) {
-    std::vector<std::pair<query::v2::Expression *, query::v2::plan::FilterInfo>> pk;
-    const auto schema = request_router_->GetSchemaForLabel(label);
-
-    std::vector<storage::v3::PropertyId> schema_properties;
-    schema_properties.reserve(schema.size());
+  bool PrimaryLabelExists(storage::v3::LabelId label) { return request_router_->IsPrimaryLabel(label); }
 
-    std::transform(schema.begin(), schema.end(), std::back_inserter(schema_properties),
-                   [](const auto &schema_elem) { return schema_elem.property_id; });
-
-    for (const auto &property_filter : property_filters) {
-      const auto &property_id = NameToProperty(property_filter.property_filter->property_.name);
-      if (std::find(schema_properties.begin(), schema_properties.end(), property_id) != schema_properties.end()) {
-        pk.emplace_back(std::make_pair(property_filter.expression, property_filter));
-      }
-    }
+  bool LabelPropertyIndexExists(storage::v3::LabelId /*label*/, storage::v3::PropertyId /*property*/) { return false; }
 
-    return pk.size() == schema_properties.size()
-               ? pk
-               : std::vector<std::pair<query::v2::Expression *, query::v2::plan::FilterInfo>>{};
+  std::vector<memgraph::storage::v3::SchemaProperty> GetSchemaForLabel(storage::v3::LabelId label) {
```

Comment:
I didn't catch this before, but both this and the function in `RequestRouter` should return `const&` to not copy the schema always.